dbms/src/Storages/Page/V3/Universal/UniversalWriteBatchImpl.h

```diff
@@ -101,6 +103,19 @@ class UniversalWriteBatch : private boost::noncopyable
         putPage(page_id, tag, buffer_ptr, data.size(), data_sizes);
     }
 
+    void putRemotePage(const UniversalPageId & page_id, UInt64 tag, const PS::V3::CheckpointLocation & data_location, PageFieldOffsetChecksums && offset_and_checksums)
```

Comment:
I think we also need to add `putRemoteExternal(const UniversalPageId & page_id, const PS::V3::CheckpointLocation & data_location)` to ingest a remote dmfile into UniPS and restore it after restart?

Reply:
I suggest put it in the later pr about FAP. Because then we can determine how to implement `putRemoteExternal`.

Reply:
OK